### PR TITLE
Election System for Counting Votes

### PR DESCRIPTION
**Pull request by lachierussell**
## Change Summary
Created the vote counting mechanism. This uses hare-clark which is the optional preferential voting mechanism used in Tasmania for a multi-position election race. The Australian Senate uses a variation of this algorithm (to support above-the-line voting) that's not necessary here.

Hare-clark is a superset of instant-run-off voting used to elect a single candidate (ie - presidential race). Running the algorithm with 1 position will result in a standard instant-run-off race.

### Change Form
- [ ] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [x] The change has tests
- [] The change has documentation

Documentation is coming but I want to merge it in so the DB and application code can be joined. I need to add more tests to check edge cases, and manually count the tests to make sure they're correct.

### CODE REVIEW

**Comment by lachierussell** [comment]
Test `Positions › should get all positions` seems to be broken for me on main too

**Review by Fil-Naj — APPROVED** [review]

**Comment by Fil-Naj** [comment]
LGTM 👍

File: src/lib/election-system/hare-clark.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      }
+      // console.log(this.candidates);
+    }
+    return elected;
+  }
+
+  nextValidPreference(v: Vote): string | undefined {
+    let next;
+    do {
+      next = v.next();
```
Comment (Drewbi) [inline src/lib/election-system/hare-clark.ts:58]: nice

**Review by Drewbi — APPROVED** [review]
